src/components/Detail/UpdateForm.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from 'react';
+
+import { trpc } from '../../utils/trpc';
+import { Button } from '../Button/Button';
+import { TextField } from '../TextField/TextField';
+import style from './UpdateForm.module.scss';
+
+type UpdateFormProps = {
+  activity: {
+    name: string;
+    amount: number;
+    unit: string;
+    timesDone: number;
+    id: number;
+  };
+};
+
+export const UpdateForm: React.FC<UpdateFormProps> = ({ activity }) => {
+  const [currentAmount, setCurrentAmount] = useState(0);
+
+  const utils = trpc.useContext();
+
+  const createActivityRecord = trpc.activityRecord.create.useMutation({
+    onSuccess: () => {
+      utils.activityRecord.invalidate();
+    },
+  });
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    createActivityRecord.mutate({
+      activityId: activity.id,
+      addedAmount: currentAmount,
+    });
+
+    setCurrentAmount(0);
+  };
+
+  return (
+    <section className={style.updateBox}>
+      <div>
+        CÍL
+        <div className={style.amountBox}>
+          <p>{activity.amount}</p>
+          <span>{activity.unit}</span>
+        </div>
+      </div>
+      <form onSubmit={handleSubmit}>
+        <div>
+          <TextField
+            label="HOTOVO"
+            name="timesDone"
+            type="number"
+            onChange={(e) => setCurrentAmount(+Number(e.target.value))}
```

Comment:
`+Number` The + is so tiny that it could be easily missed, and it is hard to read. If you want ONLY a positive number. Add validation for it with min/max. Here is what native html5 validation offers -> https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

I believe your "vyplnte vsechna pole" could be partially handled by that as well. Browsers will take care of the error messages with the:

![CleanShot 2024-01-04 at 12 36 47](https://github.com/KamilaSemotanova/tracker/assets/12488648/dc3cb087-2fbb-4ec4-b7b5-7dcd3d35c37b)

But since you have it, with the nice error message - keep it for the "required". Add the validation for min/max instead of the `+` :D

Reply:
Btw if you want to convert a negative number to a positive, I would recommend using `Math.abs`. It is much clearer what the intention was. `+` could be mistaken for implicit casting attempt and some good soul could try to rewrite it to `Number(x)`

Reply:
done :)